0.13.6 eviction notices too noisy

eed3si9n opened the pull request:
It's pretty cool that SBT now lets me know of evictions. The problem with it is that it warns so often; I think we need a way to say _I'm aware of eviction X, please don't bug me about it_.

Example, this is what `~compile` looks like now for me.

```
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.github.japgolly.fork.scalaz:scalaz-effect_sjs0.5_2.11:7.1.0-2 -> 7.1.0-3
[warn] Run 'evicted' to see detailed eviction warnings
[success] Total time: 0 s, completed 24/09/2014 5:18:43 PM
1. Waiting for source changes... (press enter to interrupt)
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.github.japgolly.fork.scalaz:scalaz-effect_sjs0.5_2.11:7.1.0-2 -> 7.1.0-3
[warn] Run 'evicted' to see detailed eviction warnings
[success] Total time: 0 s, completed 24/09/2014 5:19:01 PM
2. Waiting for source changes... (press enter to interrupt)
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.github.japgolly.fork.scalaz:scalaz-effect_sjs0.5_2.11:7.1.0-2 -> 7.1.0-3
[warn] Run 'evicted' to see detailed eviction warnings
[success] Total time: 0 s, completed 24/09/2014 5:19:04 PM
3. Waiting for source changes... (press enter to interrupt)
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.github.japgolly.fork.scalaz:scalaz-effect_sjs0.5_2.11:7.1.0-2 -> 7.1.0-3
[warn] Run 'evicted' to see detailed eviction warnings
[success] Total time: 0 s, completed 24/09/2014 5:19:06 PM
4. Waiting for source changes... (press enter to interrupt)
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.github.japgolly.fork.scalaz:scalaz-effect_sjs0.5_2.11:7.1.0-2 -> 7.1.0-3
[warn] Run 'evicted' to see detailed eviction warnings
[success] Total time: 0 s, completed 24/09/2014 5:19:08 PM
5. Waiting for source changes... (press enter to interrupt)
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.github.japgolly.fork.scalaz:scalaz-effect_sjs0.5_2.11:7.1.0-2 -> 7.1.0-3
[warn] Run 'evicted' to see detailed eviction warnings
[success] Total time: 0 s, completed 24/09/2014 5:19:10 PM
6. Waiting for source changes... (press enter to interrupt)
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn]  * com.github.japgolly.fork.scalaz:scalaz-effect_sjs0.5_2.11:7.1.0-2 -> 7.1.0-3
[warn] Run 'evicted' to see detailed eviction warnings
[success] Total time: 0 s, completed 24/09/2014 5:19:13 PM
7. Waiting for source changes... (press enter to interrupt)
```

eed3si9n commented:
Yea. It should only print it when the graph changes.
Also `7.1.0-2 -> 7.1.0-3` looks ok to evict.

jsuereth commented:
I bet it's detecting those as -SNAPSHOTs.

brunon commented:
I have the same problem - any way to configure the eviction notices?

aloiscochard commented:
Same issue here, +1 for that feature

ijuma commented:
In the meantime, you can use the following to switch off noisy eviction warnings:

`evictionWarningOptions in update := EvictionWarningOptions.default.withWarnDirectEvictions(false)`

Not ideal, of course, but necessary for sanity in my opinion.

aloiscochard commented:
@ijuma thanks mate! now that you mention it that remind me something ;-)

eed3si9n committed:
Fixes minor mistake in #1748

eed3si9n committed:
Fixes #1615. Move the invocation of eviction warning.

This moves the invocation of eviction warning so it’s called only when
actual ivy update is being performed.

eed3si9n commented:
Attached a PR. /review @jsuereth

eed3si9n committed:
'evicted' task to display all evictions
